fails to build with GCC-12 -O3 (MSYS2 + MinGW)

```cc
[2/15] Building C object src/CMakeFiles/woof.dir/z_zone.c.obj
FAILED: src/CMakeFiles/woof.dir/z_zone.c.obj
D:\msys64\mingw32\bin\cc.exe -DHAVE_FLUIDSYNTH -DMINIZ_NO_ARCHIVE_APIS -DMINIZ_NO_STDIO -DMINIZ_NO_TIME -DRANGECHECK -ID:/msys64/home/FGreffrath/woof/build/src/.. -ID:/msys64/home/FGreffrath/woof/textscreen/. -ID:/msys64/home/FGreffrath/woof/opl/. -isystem D:/msys64/mingw32/include/SDL2 -O3 -DNDEBUG -Wdeclaration-after-statement -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Wformat=2 -Wnull-dereference -Wredundant-decls -Wrestrict -Wall -Werror -Dmain=SDL_main -MD -MT src/CMakeFiles/woof.dir/z_zone.c.obj -MF src\CMakeFiles\woof.dir\z_zone.c.obj.d -o src/CMakeFiles/woof.dir/z_zone.c.obj -c D:/msys64/home/FGreffrath/woof/src/z_zone.c
D:/msys64/home/FGreffrath/woof/src/z_zone.c: In function 'Z_Malloc':
D:/msys64/home/FGreffrath/woof/src/z_zone.c:335:32: error: pointer 'block' may be used after 'free' [-Werror=use-after-free]
  335 |             (newb->next = block->next)->prev = newb;
      |                           ~~~~~^~~~~~
In function 'Z_Free',
    inlined from 'Z_Malloc' at D:/msys64/home/FGreffrath/woof/src/z_zone.c:316:10:
D:/msys64/home/FGreffrath/woof/src/z_zone.c:466:10: note: call to 'free' here
  466 |          free(block);
      |          ^~~~~~~~~~~
D:/msys64/home/FGreffrath/woof/src/z_zone.c: In function 'Z_Malloc':
D:/msys64/home/FGreffrath/woof/src/z_zone.c:327:15: error: pointer 'block' may be used after 'free' [-Werror=use-after-free]
  327 |       if(block->tag == PU_FREE && block->size >= size)   // First-fit
      |          ~~~~~^~~~~
In function 'Z_Free',
    inlined from 'Z_Malloc' at D:/msys64/home/FGreffrath/woof/src/z_zone.c:316:10:
D:/msys64/home/FGreffrath/woof/src/z_zone.c:466:10: note: call to 'free' here
  466 |          free(block);
      |          ^~~~~~~~~~~
cc1.exe: all warnings being treated as errors
```

--- a/src/z_zone.c
+++ b/src/z_zone.c
@@ -310,7 +310,7 @@ void *(Z_Malloc)(size_t size, int tag, void **user, const char *file, int line)
    do
    {
       // Free purgable blocks; replacement is roughly FIFO
-      if(block->tag >= PU_PURGELEVEL && !block->vm)
+      if(block->tag >= PU_PURGELEVEL)
       {
          start = block->prev;
          Z_Free((char *) block + HEADER_SIZE);
@@ -463,7 +463,7 @@ void (Z_Free)(void *p, const char *file, int line)
 #ifdef INSTRUMENTED
          virtual_memory -= block->size;
 #endif
-         free(block);
+//       free(block); // [FG] TODO
       }
       else
       {

--- a/src/z_zone.c
+++ b/src/z_zone.c
@@ -310,7 +310,7 @@ void *(Z_Malloc)(size_t size, int tag, void **user, const char *file, int line)
    do
    {
       // Free purgable blocks; replacement is roughly FIFO
-      if(block->tag >= PU_PURGELEVEL)
+      if(block->tag >= PU_PURGELEVEL && !block->vm)
       {
          start = block->prev;
          Z_Free((char *) block + HEADER_SIZE);

--- a/textscreen/txt_window.c
+++ b/textscreen/txt_window.c
@@ -555,12 +555,12 @@ void TXT_OpenURL(const char *url)
 #endif
 
     retval = system(cmd);
-    free(cmd);
     if (retval != 0)
     {
         fprintf(stderr, "TXT_OpenURL: error executing '%s'; return code %d\n",
             cmd, retval);
     }
+    free(cmd);
 }
 
 #endif /* #ifndef _WIN32 */

